### PR TITLE
feat(compat): Fix default CJS export, add ESM wrapper

### PR DESCRIPTION
**Pull request by jnv**
OK, so I managed to fix the backward compatibility without `default` export. I didn't manage to create CJS / ESM hybrid, but I think this is good enough.

I tested this on the `twitter-demo` project and it works fine, including type hints. But I haven't tested this on any TS or ESM-only project.

I think this should fix the minor version compatibility. In the end, I doubt it makes much sense to create CJS/ESM hybrid for this package since the whole Express / Passport.js stack is tightly coupled with Node.js ecosystem and there's not much priority on migrating to ESM there as well, but maybe someone will prove me wrong.

### CODE REVIEW

**Review by janhalama — APPROVED** [review]
🎉 you made it happen 👍

**Comment by janhalama** [comment]
I agree that we do not need  create CJS / ESM hybrid at this point.

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,14 +7,15 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## Unreleased
 
+The module was **rewritten to TypeScript.** It should be fully backwards compatible, but if you run into any issues, [please report them](https://github.com/superfaceai/passport-twitter-oauth2/issues).
```
Comment (janhalama) [inline CHANGELOG.md:10]:
```suggestion
### Changed
- The module was **rewritten to TypeScript.** It should be fully backwards compatible, but if you run into any issues, [please report them](https://github.com/superfaceai/passport-twitter-oauth2/issues).
```

I am afraid this is not valid CHANGELOG format.

Comment (janhalama) [reply CHANGELOG.md:10]: There is already `Changed` section, the suggestion can not be used.

Comment (jnv) [reply CHANGELOG.md:10]: Right. I think it is fine from the convention's point of view, but dunno if parser will accept it. So I will move the Changed section to top and add this as a first point.

Comment (janhalama) [reply CHANGELOG.md:10]: We should add validation of changelog format to the testing pipeline.

**Pull request merged**